driver/csiplugin/controllerserver.go

```diff
@@ -752,106 +922,74 @@ func (cs *ScaleControllerServer) CreateVolume(ctx context.Context, req *csi.Crea
 		}
 		conn, err := cs.getConnFromClusterID(ctx, scaleVol.ClusterId)
 		if err != nil {
-			return nil, err
+			return err
 		}
 		scaleVol.Connector = conn
 	} else {
 		scaleVol.Connector = scaleVol.PrimaryConnector
 		scaleVol.ClusterId = primaryClusterID
 	}
+	return nil
+}
 
-	if isNewVolumeType {
-		if err := cs.checkCGSupport(ctx, scaleVol.Connector); err != nil {
-			return nil, err
-		}
-	}
-	if isVolSource {
-		err = cs.validateCloneRequest(ctx, &srcVolumeIDMembers, scaleVol, primaryClusterID, volFsInfo)
+func (cs *ScaleControllerServer) checkVolTierNdSetFilesystemPolicy(ctx context.Context, scaleVol *scaleVolume, volFsInfo connectors.FileSystem_v2, volName string) error {
```

Comment:
Typo: `checkVolTierNdSetFilesystemPolicy` should be `checkVolTierAndSetFilesystemPolicy`

Reply:
addressed the change.